### DISCUSSION TITLE
Server Actions/Functions doesn't seem to be running on my NextJs app deployed on vercel.

### DISCUSSION THREAD

**Discussion by rajnish07** [vercel/next.js#64979] [Help]
### Summary

I have an app deployed to vercel. It is having a no. of Server functions which run on server. They do work when I'm running the app locally and I can see the logs for them on local server.
But when I cannot see them running(not sure fully) as neither I see the action working nor any failure logs.
Can some help to about where exactly should I check about this?, or do I need to enable anything.

### Additional information

```js
Here is the link to my app https://interviewmate.vercel.app/

When you try to run some code for the first time it sends an OTP on your mail and you need to enter the same on verify dialog to verify your account.

The email is triggered when I run this locally, but I do not see the mails coming when deployed.
FYI - I'm using nodemailer for sending mails.
```

**Comment by 2208Prathamesh** [comment]
Hi! I think this issue might be due to where your Server Actions are located:

- Server Actions must live inside the `app/` directory (not `pages/`).
- They only work when used inside components—not API routes.
- On Vercel, misplacement may silently prevent invocation—try adding `console.log("hit")` or throwing an error to test if the action is running at all.

Drop your folder structure or deployment info if you're still stuck—I’d be happy to help dig in!
